resources/template.html

```diff
@@ -142,20 +142,26 @@ <h2 class="mb-0">
 			{{- else}}
 			{{- with .Histogram}}
 			<table class="table table-striped table-bordered">
-				{{- range .Bucket}}
+                                {{- if le (len .Bucket) 0}}
 				<tr>
-					<th scope="row">Sample values &le; {{value .GetUpperBound}}</th>
-					<td>{{.GetCumulativeCount}}</td>
-				</tr>
-				{{- end}}
-				<tr>
-					<th scope="row">Total sample Count</th>
-					<td>{{.GetSampleCount}}</td>
-				</tr>
-				<tr>
-					<th scope="row">Sample Sum</th>
-					<td>{{value .GetSampleSum}}</td>
+					<td>{{.String}}</td>
```

Comment:
The problem here is that this is just the string representation of the protobuf, which is not very human-readable.

There should be code somewhere already to convert a protobuf histogram to `model.histogram.Histogram` or `model.histogram.FloatHistogram`. Maybe we can utilize it to get a display here that looks similar to what we currently get in the Prometheus UI itself?

I'm in a hurry right now, but I can help you to find this conversion code if you cannot find it easily.

Reply:
No worries, I'll go dig. Thanks for the pointer.

Reply:
So the code that does this during scraping is unfortunately too tightly coupled with the parser:
https://github.com/prometheus/prometheus/blob/980e2895a2eebb9664af104e6d4e19932cc74432/model/textparse/protobufparse.go#L173-L214

Maybe you could extract a separate `ProtoToHistogram(*dto.Histogram) (*histogram.Histogram, *histogram.FloatHistogram)` from that code (to be used there and then in the PGW, too).

Reply:
Or maybe not. In prometheus/prometheus, we use Gogo protobuf, in prometheus/pushgateway, we use the official Go protobuf library. So the dto.Histogram types are actually different.
I would just write the conversion code proto -> histogram.Histogram here in PGW for now. Depending on how long it is and where it is actually used, we can then move it somewhere else if needed.